include/envoy/stats/stats.h

```diff
@@ -113,26 +115,15 @@ class Gauge : public virtual Metric {
   virtual void sub(uint64_t amount) PURE;
   virtual uint64_t value() const PURE;
 
-  // Different approaches to importing a parent's stat value. Only used for gauges; all counters
-  // simply bring in the periodic deltas.
-  enum class CombineLogic {
-    // the default; the merged result is old+new.
-    Accumulate = Flags::LogicAccumulate,
-    // import parent value only if child stat is undefined. (So, just once.)
-    OnlyImportWhenUnusedInChild = Flags::LogicUnusedOnly,
-    // ignore parent entirely; child stat is undefined until it sets its own value.
-    NoImport = Flags::LogicNeverImport,
-  };
-
   /**
    * Returns the stat's combine logic, if known.
    */
-  virtual absl::optional<CombineLogic> cachedCombineLogic() const PURE;
+  virtual absl::optional<bool> cachedCombineLogic() const PURE;
```

Comment:
FWIW, in this case, I personally prefer an enum with 2 values, since I think it makes things a lot more readable. Thoughts?

Reply:
In this particular case, I think it's even cleaner to make a couple of function names (like the one you pointed out here) more boolean sounding. I believe the only constants ever used for this type are a `return true` and `return false`, in a function named shouldImport(), so there should be no "what does `true` mean here" issues.

Reply:
Yup :+1: